<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ParticleScriptDialog</class>
 <widget class="QDialog" name="ParticleScriptDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>731</width>
    <height>635</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="leftMargin">
    <number>2</number>
   </property>
   <property name="rightMargin">
    <number>2</number>
   </property>
   <property name="bottomMargin">
    <number>2</number>
   </property>
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tabInit">
      <attribute name="title">
       <string>Init script</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <property name="margin">
        <number>0</number>
       </property>
       <item>
        <widget class="ParticleScriptEditor" name="initScript">
         <property name="font">
          <font>
           <family>Menlo</family>
           <pointsize>12</pointsize>
          </font>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tabRender">
      <attribute name="title">
       <string>Render script</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <property name="margin">
        <number>0</number>
       </property>
       <item>
        <widget class="ParticleScriptEditor" name="renderScript">
         <property name="font">
          <font>
           <family>Menlo</family>
           <pointsize>12</pointsize>
          </font>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <layout class="QGridLayout" name="gridLayout">
     <item row="0" column="3">
      <widget class="QPushButton" name="btnSave">
       <property name="text">
        <string>Save</string>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QPushButton" name="btnLoadState">
       <property name="text">
        <string>Load from state</string>
       </property>
      </widget>
     </item>
     <item row="0" column="2">
      <widget class="QPushButton" name="btnSaveAs">
       <property name="text">
        <string>Save as</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QPushButton" name="btnLoad">
       <property name="text">
        <string>Load custom script</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ParticleScriptEditor</class>
   <extends>QPlainTextEdit</extends>
   <header>widgets/particlescripteditor</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>btnLoadState</sender>
   <signal>clicked()</signal>
   <receiver>ParticleScriptDialog</receiver>
   <slot>onLoadFromState()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>91</x>
     <y>616</y>
    </hint>
    <hint type="destinationlabel">
     <x>365</x>
     <y>317</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnLoad</sender>
   <signal>clicked()</signal>
   <receiver>ParticleScriptDialog</receiver>
   <slot>onLoadCustom()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>274</x>
     <y>616</y>
    </hint>
    <hint type="destinationlabel">
     <x>365</x>
     <y>317</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnSave</sender>
   <signal>clicked()</signal>
   <receiver>ParticleScriptDialog</receiver>
   <slot>onSave()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>638</x>
     <y>616</y>
    </hint>
    <hint type="destinationlabel">
     <x>365</x>
     <y>317</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnSaveAs</sender>
   <signal>clicked()</signal>
   <receiver>ParticleScriptDialog</receiver>
   <slot>onSaveAs()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>456</x>
     <y>616</y>
    </hint>
    <hint type="destinationlabel">
     <x>365</x>
     <y>317</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onLoadFromState()</slot>
  <slot>onSaveAs()</slot>
  <slot>onLoadCustom()</slot>
  <slot>onSave()</slot>
 </slots>
</ui>
